File: codigos_clase/practica4/CODIGO/PYTHON/Suavizado/bilateralSuavizado.py
import cv2
import numpy as np

img = cv2.imread("../../data/images/gaussian-noise.png")

# Comprobamos se se leu a imaxe
if img is None:
    print("Non poiden ler a imaxe")

# diametro para considerar entorno da vecindade
dia=15

# Valores altos mestura cores distantes
# producindo areas de cores semi-iguais
sigmaColor=80

#Valores altos permiten a influencia dos pixels alonxados
sigmaSpace=80

#Aplicamos o filtro
result = cv2.bilateralFilter(img, dia, sigmaColor, sigmaSpace)

cv2.imshow("Imaxe Orixinal", img)
cv2.waitKey(0)
cv2.imshow("Resulado do Filtro Bilateral", result)
cv2.waitKey(0)

File: codigos_clase/practica3/C++/Suavizado/bilateralSuavizado.cpp
#include <iostream>
#include <opencv2/imgproc.hpp>
#include <opencv2/highgui.hpp>

using namespace std;
using namespace cv;

int main(){
string filename = "../../../data/images/gaussian-noise.png";
Mat image = imread(filename);

//Diametro para incluir a vecindade para o filtrado
int dia=15;

// Valores elevados mesturar as cores
// para producir áreas de cores semi-guais
double sigmaColor=80;

// Valores grandes ten en conta valores de pixel moi alonxados de pixel de interes
double sigmaSpace=80;

Mat bilateralFiltered;

// Aplicamos o filtro bilatral
bilateralFilter(image, bilateralFiltered, dia,
              sigmaColor,
              sigmaSpace);

imshow("Imaxe Orixinal",image);
waitKey(0);
imshow("Resultado do suavizado bilateral",bilateralFiltered);
waitKey(0);

return 0;
}
